### PR TITLE
fix(docs): Add explanation for ldap authn edit command params.

### PR DESCRIPTION
**Pull request by nhayes**

### CODE REVIEW

**Comment by lwander** [comment]
Thanks! Should we let @stewchen review the auth-specific wording?

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/ldap/EditLdapCommand.java [hunk 1]
```diff
@@ -49,19 +49,24 @@
...
   )
   private String userSearchBase;
 
   @Parameter(
       names = "--user-search-filter",
-      description = "Placeholder"
+      description = "The filter to use when searching for a user's DN. Will search either from "
+        + "user-search-base (if specified) or root for entires matching the filter, then attempt "
+        + "to bind as that user with the login password. For example, the filter 'uid={0}' would "
+        + " apply to any user whose uid matched the user's login name."
```
Comment (dorbin) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/security/authn/ldap/EditLdapCommand.java]: I'm still not getting the difference between --user-dn-pattern and --user-search-filter. Could you explain them in terms of each other? Contrast them?

File: docs/commands.md [hunk 2]
```diff
@@ -4691,9 +4691,9 @@ hal config security authn ldap edit [parameters]
  * `--deployment`: If supplied, use this Halyard deployment. This will _not_ create a new deployment.
  * `--no-validate`: (*Default*: `false`) Skip validation.
  * `--url`: ldap:// or ldaps:// url of the LDAP server
- * `--user-dn-pattern`: Placeholder...uid={0},ou=users
- * `--user-search-base`: Placeholder...
- * `--user-search-filter`: Placeholder
+ * `--user-dn-pattern`: The pattern for finding a user's DN using simple pattern matching. For example, the pattern 'uid={0},ou=users' finds the user's DN by substituting the user's login name in the supplied pattern.
+ * `--user-search-base`: The part of the directory tree under which user searches should be performed.
+ * `--user-search-filter`: The filter to use when searching for a user's DN. Will search either from user-search-base (if specified) or root for entires matching the filter, then attempt to bind as that user with the login password. For example, the filter 'uid={0}' would  apply to any user whose uid matched the user's login name.
```
Comment (dorbin) [inline docs/commands.md]: Should this file be edited if it's automatically generated?

Comment (lwander) [reply docs/commands.md]: It was automatically generated based on the changes to halyard

File: docs/commands.md [hunk 3]
```diff
@@ -4691,9 +4691,9 @@ hal config security authn ldap edit [parameters]
  * `--deployment`: If supplied, use this Halyard deployment. This will _not_ create a new deployment.
  * `--no-validate`: (*Default*: `false`) Skip validation.
  * `--url`: ldap:// or ldaps:// url of the LDAP server
- * `--user-dn-pattern`: Placeholder...uid={0},ou=users
- * `--user-search-base`: Placeholder...
- * `--user-search-filter`: Placeholder
+ * `--user-dn-pattern`: The pattern for finding a user's DN using simple pattern matching. For example, if your LDAP server has the URL ldap://mysite.com/dc=spinnaker,dc=org, and you have the pattern 'uid={0},ou=members', 'me' will map to a DN uid=me,ou=members,dc=spinnaker,dc=org. If no match is found, will try to find the user using user-search-filter, if set.
+ * `--user-search-base`: The part of the directory tree under which user searches should be performed.
```
Comment (stewchen) [inline docs/commands.md]: Also add "If user-search-base isn't supplied, the search will be performed from the root."

Comment (nhayes) [reply docs/commands.md]: Done.

**Pull request merged**